honeybee_radiance_command/gensky.py

```diff
@@ -0,0 +1,257 @@
+                'Time zone must a three letter string from the following'
+                ' options %s.' % (time_zones)
+            )
+
+    @property
+    def solar_time(self):
+        """Whether solar time is requested."""
+        return self._solar_time
+
+    @solar_time.setter
+    def solar_time(self, value):
+        if not value:
+            self._solar_time = None
+        elif isinstance(value, bool):
+            self._solar_time = value
+        else:
+            raise ValueError('Solar time only accepts True or False as a value.')
+
+    @property
+    def input(self):
+        """Input string for gensky."""
+
+        if not self.month and not self.day and not self.time:
+            self._input = None
+            return self._input
+
+        # Only month, day and time provided as arguments
+        elif not (self.time_zone and self.solar_time):
+            self._input = '%s %s %s' % (str(self.month), str(self.day), str(self.time))
+            return self._input
+
+        # Time zone is provided
+        elif self.time_zone and not self.solar_time:
+            self._input = '%s %s %s%s' % (str(self.month), str(self.day),
+                                          str(self.time), self.time_zone)
+            return self._input
+
+        # Local solar time requested
+        elif self.solar_time and not self.time_zone:
+            self._input = '+%s %s %s' % (str(self.month), str(self.day),
+                                         str(self.time))
+            return self._input
+
+        # Time zone and solar time both requested
+        elif self.time_zone and self.solar_time:
+            self._input = '+%s %s %s%s' % (str(self.month), str(self.day),
+                                           str(self.time), self.time_zone)
+            return self._input
+
+    @classmethod
+    def from_ang(cls, angles, options=None):
+        """Create a Gensky command using sun altitude and azimuth angles in degrees.
+
+        The altitude is measured in degrees above the horizon, and the azimuth is
+        measured in degrees west of South.
+
+        Args:
+            angles: A tuple of altitude and azimuth angles.
+            options: Command options. It will be set to Radiance default values if not
+                provided by user.
+        """
+        if not options:
+            options = GenskyOptions()
+
+        if angles:
+            options.ang = tuple_with_length(angles, length=2)
+
+        return cls(options=options)
+
+    def to_radiance(self, stdin_input=False):
+        """Command in Radiance format.
+
+        Args:
+            stdin_input: A boolean that indicates if the input for this command
+                comes from stdin. This is for instance the case when you pipe the input
+                from another command (default: False).
+        """
+        # If form_ang is not used, validate arguments
+        if not self.options.ang.is_set:
+            self.validate()
```

Comment:
Shouldn't you do a check to ensure both of them are not provided? Say a case that user sets angle in Radiance options and also provides day, month and time.

Reply:
Yes. I created a check.